wordpress-mobile/WordPress-iOS
# Issue/8933 Sync Tracks Opt Out setting

Fixes #8933.

This PR integrates the existing analytics tracking toggle in App Settings with the new `tracks_opt_out` account setting.

* AccountSettings has been updated to fetch and save the new tracks opt out value.
* The tracks UI toggle in `AppSettingsViewController` should now reflect this value, and also update the remote value when you toggle it.
* The app should correctly enable / disable tracks on startup.
* I've switched the `isTrackingUsage` setting in `WPAppAnalytics` over to a new `userHasOptedOut` setting that closer matches the remote setting. We migrate the value over in `initializeOptOutTracking`.

**To test:**

* Check that a fresh app install has tracks defaulted to on, and that tracking starts.
* Toggle the setting, and ensure that the next time the account settings are fetched they contain the new value for the `tracks_opt_out` key.
* Opt out of tracks using the setting, and once the value has synced remotely do a fresh install of the app. Log in with the same user, and check the setting is restored.

cc @koke @elibud
